### DISCUSSION TITLE
Bring back CLS for remote images

### DISCUSSION THREAD

**Discussion by macmillen** [withastro/roadmap#711] [Proposal]
### Body

# Summary
In Astro 2 with the `@astrojs/image` package it was possible to process and optimize remote images. That way you could prevent CLS by using the `fit={"cover"}` property. Unfortunately this is not possible anymore in Astro 3 by using `astro:assets`.

## Astro 2 way (optimized):
``` jsx
<Image
  src={cover}
  width={64}
  aspectRatio={64 / 96}
  fit={"cover"}
/>
```

## Astro 3 way (not optimized):
``` jsx
<Image
  src={cover}
  width={64}
  height={96}
/>
```
 
# Goals
Optimize remote images and prevent CLS.

**Comment by Princesseuh** [comment]
You can optimize remote images with `astro:assets`. You just have to make sure the remote URLs are authorized in your config: https://docs.astro.build/en/guides/images/#authorizing-remote-images

**Reply by macmillen** [reply]
I did that but there is still the problem that it doesn't create the image with the correct size

``` js
export default defineConfig({
  image: {
    remotePatterns: [{ protocol: "https" }],
  },
});
```

When I use the <Image /> component like this:
``` jsx
<Image
  src={cover}
  width={64}
  height={96}
/>
```

It creates an image with 64x85 instead of 64x96 unlike in the original implementation where I could set `fit={"cover"}` (see my first post)

**Reply by Princesseuh** [reply]
Ah yes, you mean it doesn't crop images. This is intended at this time. It's unrelated to CLS or remote images, the default image services don't have support for cropping right now

**Reply by macmillen** [reply]
Ah got it, I completely misunderstood what issue was. sorry for that :sweat_smile: 
But doesn't it still use Sharp for image processing as in the previous implementation?

**Reply by Princesseuh** [reply]
It does, but cropping was intentionally omitted from this first version of `astro:assets` for multiple reasons:
- Not every image service support it
- Multiple users were confused by the reverse issue, where it would crop their images and they didn't understand why

We may bring it back in the future though, I know on a personal level I'd love to see it again, at least

**Reply by Princesseuh** [reply]
FWIW, implementing an image service that does support cropping is fairly easy to do in user-land, see the documentation here: https://docs.astro.build/en/reference/image-service-reference/

It's essentially just a slightly different `transform` hook than the built-in one here: <https://github.com/withastro/astro/blob/main/packages/astro/src/assets/services/sharp.ts>

**Reply by macmillen** [reply]
Cool, thank you so much! :smiley:
